src/fuzzfetch/fetch.py

```diff
@@ -831,103 +949,11 @@ def from_args(cls, args=None, skip_dir_check=False):
-
-        test_group = parser.add_argument_group('Test Arguments')
-        test_group.add_argument('--tests', nargs='+', metavar='', choices=cls.TEST_CHOICES,
-                                help=('Download tests associated with this build. Acceptable values are: ' +
-                                      ', '.join(cls.TEST_CHOICES)))
-        test_group.add_argument('--full-symbols', action='store_true',
-                                help='Download the full crashreport-symbols.zip archive.')
-
-        misc_group = parser.add_argument_group('Misc. Arguments')
-        misc_group.add_argument('-n', '--name',
-                                help='Specify a name (default=auto)')
-        misc_group.add_argument('-o', '--out', default=os.getcwd(),
-                                help='Specify output directory (default=.)')
-        misc_group.add_argument('--dry-run', action='store_true',
-                                help="Search for build and output metadata only, don't download anything.")
-
-        near_group = parser.add_argument_group('Near Arguments', "If the specified build isn't found, iterate over " +
-                                               "builds in the specified direction")
-        near_args = near_group.add_mutually_exclusive_group()
-        near_args.add_argument('--nearest-newer', action='store_const', const=cls.BUILD_ORDER_ASC, dest='nearest',
-                               help="Search from specified build in ascending order")
-        near_args.add_argument('--nearest-older', action='store_const', const=cls.BUILD_ORDER_DESC, dest='nearest',
-                               help="Search from the specified build in descending order")
-
-        args = parser.parse_args(args=args)
-
-        if re.match(r'(\d{4}-\d{2}-\d{2}|[0-9A-Fa-f]{12}|[0-9A-Fa-f]{40}|latest)$', args.build) is None:
-            # this is a custom build
-            # ensure conflicting options are not set
-            if args.branch is not None:
-                parser.error('Cannot specify --build namespace and branch argument: %s' % args.branch)
-            if args.debug:
-                parser.error('Cannot specify --build namespace and --debug')
-            if args.asan:
-                parser.error('Cannot specify --build namespace and --asan')
-            if args.tsan:
-                parser.error('Cannot specify --build namespace and --tsan')
-            if args.fuzzing:
-                parser.error('Cannot specify --build namespace and --fuzzing')
-            if args.coverage:
-                parser.error('Cannot specify --build namespace and --coverage')
-            if args.valgrind:
-                parser.error('Cannot specify --build namespace and --valgrind')
+        args = FetcherArgs().parse_args(args)
 
         # do this default manually so we can error if combined with --build namespace
         # parser.set_defaults(branch='central')
-        elif args.branch is None:
+        if args.branch is None:
```

Comment:
This is broken for `--build namespace`. If we're passed a namespace, `args.branch` must be `None`, we assert that it is in `sanity_check()`.

Part 5/5 of suggestion for above.
```suggestion
        if not parser.build_is_ns and args.branch is None:
```